Overhaul of settings

pgilmorepf opened the pull request:
Remove duplication and bugs involved with lack of consistency between settings variables.

Primary:
Remove PlayFabSettings static variables: TitleId, DevSecretKey, and ProductionEnvironmentURL
Redirect all usage to PlayFabRuntimeSettings which is a Unreal supported singleton that is much more flexible, and allows direct customer control.

Removing entirely:
VerticalName, DevelopmentEnvironmentURL, and related variables & methods
Lots of relay crud methods for TitleId, DevSecretKey, & ProdUrl above. (easier to access RuntimeSettings Directly)

After that, it's all just dominoes: Things we must change/fix to stupport the above.
This PR (temporarily?) contains both the SDK changes and the template changes.

pgilmorepf committed:
target

pgilmorepf committed:
Single location TitleId, DevSecretKey, and ProductionEnvironmentURL

pgilmorepf committed:
Header import order.

pgilmorepf committed:
Restore the Project Setting registration

pgilmorepf committed:
allow full overrides for prodUrl

pgilmorepf committed:
Generator changes.

TODO: Fix GetEntityToken stuff.
TODO: Fix Post login set-context lines

pgilmorepf committed:
Templates to build new destination

toddbellMSFT commented on 4.23/ExampleProject/Plugins/PlayFab/Source/PlayFabCommon/Private/PlayFabCommon.cpp:
```diff
@@ -27,32 +27,20 @@ class FPlayFabCommonModule : public IPlayFabCommonModuleInterface
     void UnregisterSettings();
 
-    /** Callback for when the settings were saved. */
-    bool HandleSettingsSaved();
```
seems like we remove a feature with this, but I guess it also decreases complexity if it wasn't being used (It looks like it is not labeled public, so I'm assuming it is not customer facing)

toddbellMSFT commented on 4.23/ExampleProject/Plugins/PlayFab/Source/PlayFabCpp/Private/Core/PlayFabAuthenticationAPI.cpp:
```diff
@@ -74,7 +64,7 @@ void UPlayFabAuthenticationAPI::OnGetEntityTokenResult(FHttpRequestPtr HttpReque
     {
         if (outResult.EntityToken.Len() > 0)
-            PlayFabSettings::SetEntityToken(outResult.EntityToken);
+            // PlayFabSettings::SetEntityToken(outResult.EntityToken);
```
Should this commented line be removed entirely?

toddbellMSFT commented on 4.23/ExampleProject/Plugins/PlayFab/Source/PlayFabCpp/Private/Core/PlayFabAuthenticationInstanceAPI.cpp:
```diff
@@ -116,7 +115,7 @@ void UPlayFabAuthenticationInstanceAPI::OnGetEntityTokenResult(FHttpRequestPtr H
     {
         if (outResult.EntityToken.Len() > 0)
-            this->GetOrCreateAuthenticationContext()->SetEntityToken(outResult.EntityToken);
+            // context->SetEntityToken(outResult.EntityToken);
```
This may be odd. How will this compile? Will the line below count as a part of the if check? Or does this if check for something and then ... ?

pgilmorepf committed:
New generator target

The generator caught a bunch of stuff I missed, so this gets it all compiling again.

pgilmorepf committed:
intermediate target checkin

todo fix callback contexts.
maybe fix some generator stuff.

pgilmorepf committed:
updating settings.

pgilmorepf committed:
Not quite right context fixes. Compiles and runs, but sets the wrong context.

pgilmorepf committed:
fixed instance contexts.

pgilmorepf committed:
Fixing templates for instance contexts.

pgilmorepf committed:
Adding an upgrade guide for the next Unreal publish.
